src/hooks/useValidation/validators.js

```diff
@@ -252,6 +252,22 @@ export const validateLocation = ({ marcRecords, locations }, rule) => {
 
   return undefined;
 };
+export const validateSubfieldValueMatch = (context, rule) => {
+  const { marcRecords } = context;
+
+  if (rule.ignore?.(context)) {
+    return undefined;
+  }
+
+  const marcRecord = marcRecords.find(record => record.tag.match(rule.tag));
+  const subfieldValue = getContentSubfieldValue(marcRecord?.content)[rule.subfield]?.[0];
+
+  if (!subfieldValue.match(rule.pattern(context))) {
+    return rule.message();
+  }
+
+  return undefined;
+};

```

Comment:
I am _not_ suggesting that you rewrite this function, especially given that it follows the validation pattern of all the other validation functions here, but all these functions feels like they are written backwards to me. Instead of logic that checks for errors, validation functions should check whether data is valid. That is, instead of
```
if (data has error) return error
else return success
```
they should operate like this:
```
if (data is valid) return success
else return error
```
Often, there is usually only one way for data to be valid, but many of ways for it to be invalid, so testing that (usually) narrower route to success is a better option. If you find yourself starting a new validation system, please consider this more defensive approach.